Fix #18

NFRCG opened the pull request:
This change should fix #18. 

There was a regression here: https://github.com/promcteam/BossShopPro/commit/615d327b8b9a1eb909073dc22787ab6a04090fcb#diff-ad112dcac44cfd0b956233154873661e69c

This should be tested further before merge. This method is frequently accessed. Build with this fix is attached in the zip file.
[PR.zip](https://github.com/promcteam/BossShopPro/files/10714497/PR.zip)

Travja commented:
Ah interesting. Yeah. That wouldn't be checking all the item data parts and just cutting out early. Good catch.

Travja reviewed (approved):
I'm going to go ahead and approve this. I don't see any reason why this shouldn't work. At bare minimum, it's not going to make things worse 😂

FunkyNico commented:
Hello, i was testing this PR build provided here. I have a price type that is an item (glowstone dust) and has a certain name and lore, but BSP don't recognize it with this PR build, but in fact it doesn't accept just common glowstone dust also.

https://pastebin.ubuntu.com/p/ByxsVXt4Gq/

NFRCG commented:
> Hello, i was testing this PR build provided here. I have a price type that is an item (glowstone dust) and has a certain name and lore, but BSP don't recognize it with this PR build, but in fact it doesn't accept just common glowstone dust also.
> 
> https://pastebin.ubuntu.com/p/ByxsVXt4Gq/

I tested the example provided and fixed areas where the comparison was failing. I've updated the attached build with these fixes.

FunkyNico commented:
> > Hello, i was testing this PR build provided here. I have a price type that is an item (glowstone dust) and has a certain name and lore, but BSP don't recognize it with this PR build, but in fact it doesn't accept just common glowstone dust also.
> > https://pastebin.ubuntu.com/p/ByxsVXt4Gq/
> 
> I tested the example provided and fixed areas where the comparison was failing. I've updated the attached build with these fixes.

I have tried the updated PR build and now it's working perfectly. It's detects the right item with the provided name and lore.
